eeps/eep-0061.md

```diff
@@ -0,0 +1,255 @@
+    Author: Ilya Klyuchnikov <ilya(dot)klyuchnikov(at)gmail(dot)com>
+    Status: Draft
+    Type: Standards Track
+    Created: 08-Mar-2023
+    Post-History:
+****
+EEP 61: Builtin dynamic type
+----
+
+Abstract
+========
+
+This EEP proposes a new builtin type `dynamic()` (in addition to existing
+builtin type `term()`) to make Erlang friendly to gradual type checking.
+Having a special dynamic type for gradual typing is a widely adopted industrial
+solution.
+
+Gradual typing and dynamic type
+========
+
+Erlang is one of many programming languages which started as pure dynamically
+typed languages with no static type information but then got extensions
+(or dialects) to specify type information (type hints, specs) in source code.
+
+Erlang got extended with specs through
+[EEP 8](https://www.erlang.org/eeps/eep-0008)
+(Types and function specifications).
+
+The current documentation for types and specs can be found at
+<https://www.erlang.org/doc/reference_manual/typespec.html>.
+
+Beyond being used for documentation purposes, types and specs can be used by
+tooling for static analysis. The existing tools include:
+
+- [dialyzer](https://www.erlang.org/doc/man/dialyzer.html)
+- [eqWAlizer](https://github.com/whatsapp/eqwalizer)
+
+Dialyzer uses type specs in the paradigm of success typing (documented in
+[Pull Request 6281](https://github.com/erlang/otp/pull/6281) "Document
+the meaning of specs" and in the original
+[paper](https://dl.acm.org/doi/10.1145/1292520.1292523) "A language
+for specifying type contracts in Erlang and its interaction
+with success typings"). It is worth to note that dialyzer handles specs
+very differently from traditional type checkers.
+
+eqWAlizer uses type specs to perform type checking in a traditional manner.
+
+As of now, the main industrial approach to adopt or retrofit static typing
+to dynamically typed languages is through gradual typing
+(<https://en.wikipedia.org/wiki/Gradual_typing>,
+<http://samth.github.io/gradual-typing-bib>).
+Gradual typing features a special type `dynamic` or `?`, which allows to mix
+untyped (or dynamically typed) and statically typed code in a user-friendly
```

Comment:
minor typo: `in an user-friendly`

Reply:
See https://dictionary.cambridge.org/dictionary/english/user (section "More examples"): "a user"

Reply:
you are right, sorry about that